src/ethereum/ethPlugin.js

```diff
@@ -219,7 +220,79 @@ export class EthereumPlugin extends CurrencyPlugin {
       }
       return edgeParsedUriToken
     }
-    return edgeParsedUri
+
+    // Parse according to EIP-681
+    if (prefix === 'pay') {
+      const targetAddress = address
+      const functionName = parsedUri.pathname.split('/')[1]
+      const parameters = parsedUri.query
+
+      // Handle contract function invocations
+      // This is a very important measure to prevent accidental payment to contract addresses
+      switch (functionName) {
+        // ERC-20 token transfer
+        case 'transfer': {
+          const publicAddress = parameters.address ?? ''
+          const contractAddress = targetAddress ?? ''
+          const nativeAmount =
+            parameters.uint256 != null
+              ? biggyScience(parameters.uint256)
```

Comment:
`parseFloat(parameters.uint256).toFixed(DECIMAL_NUMBERS)`

Reply:
There may not be enough precision in IEEE 754